### PR TITLE
fetch top gems from redis instead of postgres

### PR DESCRIPTION
**Pull request by farukaydin**
fixes https://github.com/rubygems/rubygems.org/issues/947

cc // @sferik @indirect

### CODE REVIEW

File: test/unit/rubygem_test.rb [hunk 1]
```diff
@@ -683,5 +683,32 @@ class RubygemTest < ActiveSupport::TestCase
...
+      end
+
+      should "by postgres" do
+        assert_equal [@thin, @rake, @json, @thor, @rack], Rubygem.most_downloaded_by_db(5)
+      end
+
+      should "by redis" do
+        assert_equal [@thin, @rake, @json, @thor, @rack], Rubygem.most_downloaded_by_redis(5)
+      end
+    end
```
Comment (sferik) [inline test/unit/rubygem_test.rb]: I would trust these tests more if the ranking was different for Redis vs. Postgres.

File: app/models/rubygem.rb [hunk 2]
```diff
@@ -82,6 +82,23 @@ def self.monthly_short_dates
     monthly_dates.map { |date| date.strftime("%m/%d") }
   end
 
+  def self.most_downloaded_by_db(limit=10)
```
Comment (sferik) [inline app/models/rubygem.rb]: The default `limit` for the `downloaded` method is 5, but here (and below) it’s 10. It seems odd to me that they would be different. What’s the reason for that?

Comment (farukaydin) [inline app/models/rubygem.rb]: @sferik stats controller have ```Rubygem.downloaded(10)``` and this is the only place that calls Rubygem.downloaded. So we can just use limit parameter as 10 but you're right I should also update downloaded method limit parameter with 10. Sounds good?

Comment (sferik) [inline app/models/rubygem.rb]: :+1:

**Comment by farukaydin** [comment]
@sferik I fixed problems like you mentioned in line comments. If everything is good, I can rebase it.

File: test/unit/rubygem_test.rb [hunk 3]
```diff
@@ -496,7 +496,7 @@ class RubygemTest < ActiveSupport::TestCase
     end
 
     should "only latest downloaded versions" do
-      assert_equal [@thin, @rake, @json, @thor, @rack],        Rubygem.downloaded
+      assert_equal [@thin, @rake, @json, @thor, @rack],        Rubygem.downloaded(5)
```
Comment (farukaydin) [inline test/unit/rubygem_test.rb]: @sferik I'm not sure but I think we can leave off tests parameters as 5 for easiness.

**Comment by sferik** [comment]
Looks good to me now. Would anyone else like to review or can I merge it?

cc: @indirect, who opened the original issue that this fixes.

**Comment by farukaydin** [comment]
@sferik I guess there will be no other review.

**Comment by sferik** [comment]
@farukaydin Okay, I’ve merged it! Thanks for the patch.

**Comment by farukaydin** [comment]
thanks

**Pull request merged**